src/store/initial_state.js

```diff
@@ -45,7 +45,10 @@ const state = {
             currentFocusedPostId: '',
             messagesHistory: {
                 messages: [],
-                index: {}
+                index: {
+                    post: -1,
+                    comment: -1
+                }
             }
```

Comment:
I Changed the initial state here to reflect reset_state.

Reason, why the tests fails, is because of - this action https://github.com/mattermost/mattermost-redux/blob/ae530a32d0d4c5b0f3c8200f7d7890439e6cf9ca/test/actions/posts.test.js#L1462

Which reduces https://github.com/mattermost/mattermost-redux/blob/ae530a32d0d4c5b0f3c8200f7d7890439e6cf9ca/src/reducers/entities/posts.js#L367 to

 returns existing state as `state.index` exists so, the assertion would not have a value defined. 

I could change assertion as well instead of changing initial state.  Let me know if you want me to change.